project-addons/flask_middleware_connector/events/invoice_events.py

```diff
@@ -71,18 +70,22 @@ def delay_write_invoice(session, model_name, record_id, vals):
     up_fields = ["number", "client_ref", "date_invoice", "state", "partner_id",
                  "date_due", "subtotal_wt_rect", "subtotal_wt_rect"]
 
-    if invoice.partner_id and invoice.partner_id.web:
-        if vals.get('state', False) == 'open':
-            export_invoice.delay(session, model_name, record_id)
-        elif vals.get('state', False) == 'paid':
-            update_invoice.delay(session, model_name, record_id)
-        elif vals.get('state', False) == 'cancel':
-            unlink_invoice(session, model_name, record_id)
+    if invoice.partner_id and invoice.commercial_partner_id.web:
+        job = session.env['queue.job'].search(['&', ('func_string', 'like', '%' + str(invoice.id) + '%'), ('model_name', '=', model_name)])
+        if job:
+            if vals.get('state', False) == 'open' and 'unlink_invoice' in job[0].func_string:
+                export_invoice.delay(session, model_name, record_id, priority=5)
+            elif vals.get('state', False) == 'paid':
+                update_invoice.delay(session, model_name, record_id, priority=10)
+            elif vals.get('state', False) == 'cancel' and 'unlink_invoice' not in job[0].func_string:
```

Comment:
vals.get('state', False) == 'cancel' and job and 'unlink_invoice' not in job[0].func_string mejor